inc/search.class.php

```diff
@@ -522,6 +501,31 @@ static function prepareDatasForSearch($itemtype, array $params, array $forcedisp
       return $data;
    }
 
+   /**
+    * Parse the $criteria array to return a searchoption_id => value array
+    *
+    * @param array $criteria
+    *
+    * @return array
+    */
+   public static function getConditionSearchOptionsIds(array $criteria): array {
+      $condition_search_options_ids = [];
+      foreach ($criteria as $crit) {
+         if (isset($crit['criteria'])) {
+            // Recursion, parse child criteria
+            $child_criteria = self::getConditionSearchOptionsIds($crit['criteria']);
+            foreach (array_keys($child_criteria) as $child_crit) {
+               $condition_search_options_ids[$child_crit] = $child_crit['value'] ?? true;
```

Comment:
If `$child_crit` is a key of `$child_criteria`, then it cannot be an array, and `$child_crit['value']` is invalid.